src/typeChecker/typeErrorChecker.ts

```diff
@@ -208,23 +212,38 @@ function typeCheckAndReturnType(node: tsEs.Node, context: Context, env: TypeEnvi
       return typeCheckAndReturnArrowFunctionType(node, context, env)
     }
     case 'FunctionDeclaration':
-      // Only identifiers are used as function params in Source
-      const params = node.params as tsEs.Identifier[]
+      if (node.id === null) {
+        // Block should not be reached since node.id is only null when function declaration
+        // is part of `export default function`, which is not used in Source
+        throw new TypecheckError(node, 'Function declaration should always have an identifier')
+      }
```

Comment:
For documentation, this can be done because `FunctionDeclaration` AST nodes without a name are banned in https://github.com/source-academy/js-slang/commit/6ff73b19490239486aba533deaf609c5ccdfeabf, although it has not been merged into `master` yet as it is part of a larger PR to add local module imports.